core/src/main/scala/kafka/log/Log.scala

```diff
@@ -287,6 +287,13 @@ class Log(@volatile var dir: File,
     // The earliest leader epoch may not be flushed during a hard failure. Recover it here.
     _leaderEpochCache.truncateFromStart(logStartOffset)
 
+    // If the message format has been downgraded, we do not want to use the cache for truncation
```

Comment:
Would it be useful to add a debug log which prints out the loaded `_leaderEpochCache#epochEntries`?